api/src/routes/protected/classroom.ts

```diff
@@ -0,0 +1,147 @@
+import { FastifyPluginCallback } from 'fastify';
+import { Type } from '@fastify/type-provider-typebox';
+import { normalizeChallenges } from '../../utils/normalize.js';
+
+/**
+ * Fastify plugin for classroom-related protected routes.
+ * Provides endpoint for retrieving user data for classrooms.
+ * @param fastify - The Fastify instance.
+ * @param _options - Plugin options (unused).
+ * @param done - Callback to signal plugin registration is complete.
+ */
+export const classroomRoutes: FastifyPluginCallback = (
+  fastify,
+  _options,
+  done
+) => {
+  // Endpoint to retrieve a user's ID from a user's email.
+  fastify.post(
+    '/api/protected/classroom/get-user-id',
+    {
+      schema: {
+        body: Type.Object({
+          email: Type.String({ format: 'email' })
+        }),
+        response: {
+          200: Type.Object({
+            userId: Type.String()
+          }),
+          404: Type.Object({
+            error: Type.String()
+          })
+        }
+      }
+    },
+    async (request, reply) => {
+      const { email } = request.body as { email: string };
+
+      try {
+        // Find the user by email
+        const user = await fastify.prisma.user.findFirst({
+          where: { email },
+          select: {
+            id: true,
+            isClassroomAccount: true
+          }
```

Comment:
```suggestion
          where: { email, isClassroomAccount: true },
          select: { id: true }
```
i.e. select only what you're interested in, rather than getting everything and filtering.